dali/test/python/test_backend_impl_gpu.py

```diff
@@ -216,3 +217,51 @@ def check_squeeze(shape, dim, in_layout, expected_out_layout):
              (0, (1, 5, 1), "ABC", "BC"),
              (None, (3, 5, 1), "ABC", "AB")]:
         yield check_squeeze, shape, dim, in_layout, expected_out_layout
+
+
+# Those tests verify that the Tensor[List]Cpu/Gpu created in Python in a similar fashion
+# to how ExternalSource for samples operates keep the data alive.
+
+# The Tensor[List] take the pointer to data and store the reference to buffer/object that owns
+# the data to keep the refcount positive while the Tensor[List] lives.
+# Without this behaviour there was observable bug with creating several temporary
+# buffers in the loop and DALI not tracking references to them
+
+def test_tensor_cpu_from_numpy():
+    def create_tmp(idx):
+        a = np.full((4, 4), idx)
+        return tensors.TensorCPU(a, "")
+    out = [create_tmp(i) for i in range(4)]
+    for i, t in enumerate(out):
+        np.testing.assert_array_equal(np.array(t), np.full((4, 4), i))
+
+def test_tensor_list_cpu_from_numpy():
+    def create_tmp(idx):
+        a = np.full((4, 4), idx)
+        return tensors.TensorListCPU(a, "")
+    out = [create_tmp(i) for i in range(4)]
+    for i, tl in enumerate(out):
+        for j in range(4):
+            np.testing.assert_array_equal(np.array(tl[j]), np.full(tl[j].shape(), i))
+        np.testing.assert_array_equal(tl.as_array(), np.full((4, 4), i))
```

Comment:
Numpy one can go just to test_backend_impl.py.
I wonder if it is worth to create a similar test for DLPack as well (we have tests for DLPack in test_backend_impl_torch_dlpack.py) but not for this scenario.

Reply:
Added tests for dlpack. I would prefer to keep them grouped and not spread in several files.

Reply:
Fair point